app/services/data_exports/csv/invoices.rb

```diff
@@ -6,36 +6,24 @@
 module DataExports
   module Csv
     class Invoices < BaseService
-      DEFAULT_BATCH_SIZE = 100
-
       extend Forwardable
 
-      def initialize(data_export:, serializer_klass: V1::InvoiceSerializer, output: Tempfile.create)
-        @data_export = data_export
+      def initialize(data_export_part:, serializer_klass: V1::InvoiceSerializer)
+        @data_export_part = data_export_part
         @serializer_klass = serializer_klass
-        @output = output
-        @batch_size = DEFAULT_BATCH_SIZE
+        super
       end
 
       def call
-        ::CSV.open(output, 'wb', headers: true) do |csv|
-          csv << headers
-
-          invoices.find_each(batch_size:).lazy.each do |invoice|
+        result.csv_lines = ::CSV.generate(headers: false) do |csv|
+          invoices.each do |invoice|
             csv << serialized_invoice(invoice)
           end
         end
-
-        output.rewind
```

Comment:
isn't needed anymore to rewind the csv output?

Reply:
no, it's a string now :)